lldb/source/Plugins/Process/mach-core/ProcessMachCore.cpp

```diff
@@ -652,7 +652,7 @@ size_t ProcessMachCore::ReadMemory(addr_t addr, void *buf, size_t size,
                                    Status &error) {
   // Don't allow the caching that lldb_private::Process::ReadMemory does since
   // in core files we have it all cached our our core file anyway.
-  return DoReadMemory(addr, buf, size, error);
+  return DoReadMemory(FixAnyAddress(addr), buf, size, error);
```

Comment:
I'm tempted to move this up into `lldb/include/lldb/Target/PostMortemProcess.h` but not all of them fix addresses yet, and it's better changes like that are done explicitly by people who use those classes. If we get to a point where everyone is fixing the address then we can pull this up.